### PR TITLE
♻️ Refactor Email Templates (#761)

### PR DESCRIPTION
**Pull request by mistryrn**
## Description of Changes

Refactors the Clinician Invite and Complete Account Reminder email templates to remove logic from the routers, and adds basic usage of MJML as a starting point for further templating/styling purposes.

### Email Service
- Refactor all email templating and sending logic out of the `/invites` and `/reminders` routers, into the dedicated `/emails` directory
- Convert the plain HTML email templates to MJML
- Refactor the email templates to use a basic layout and some basic examples of reusable components (header, footer) as an example for the future email templating/styling tickets.

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [ ] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] [Axe DevTools](https://www.deque.com/axe/browser-extensions/) "Full Page Scan" of updated pages passes without introducing new Critical or Serious issues (UI changes only)
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/email-service/src/emails/common.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	return `
+		<mj-section background-color="#231F20">
+			<mj-column>
+				<mj-text color="#FFFFFF">&copy; ${new Date().getFullYear()} OHCRN. All rights reserved.</mj-text>
+			</mj-column>
+		</mj-section>
+	`;
+};
+
+export const basicLayout = (content: string) => {
```
Comment (mistryrn) [inline apps/email-service/src/emails/common.ts:40]: As mentioned during demo, this is a very simple example to show how we can compose MJML emails using "components" (functions that return MJML template strings).

Feel free to refactor/reorganize as you see fit when working on the styling of the templates 🙂

File: apps/email-service/src/emails/common.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+export const header = () => {
+	return `
+		<mj-section>
+			<mj-column>
+				<mj-image src="https://ohcrn.ca/wp-content/uploads/2023/03/oicr-ohcrn-logo-outlines-1.png" alt="logo" width="550px" />
```
Comment (mistryrn) [inline apps/email-service/src/emails/common.ts:24]: As called out during demo, currently using the main OHCRN site for images. When working on the styling tickets, we'll likely want to do the following:
1. Gather all email image assets from Figma
2. Ensure image assets are optimized with something like ImageOptim: https://imageoptim.com/mac
3. Rename assets to begin with the prefix `email_`
4. Package and send all images to whoever has Admin access for the OHCRN main site to upload and return URLs

File: apps/email-service/src/routers/invites.ts [hunk 3]
```diff
@@ -70,60 +70,21 @@ router.post(
...
+				}
+				case 'SYSTEM_ERROR': {
+					logger.error(
+						'POST /invites/participant',
+						'Failed to send Participant Clinician Invite Email.',
+						result.message,
+					);
+					return res.status(500).json(ServerErrorResponse());
+				}
+			}
```
Comment (mistryrn) [inline apps/email-service/src/routers/invites.ts:87]: Routers are now following the same pattern we've been using across our other apps and services:
- make "service" function call to handle any logic, perform required action
- return response based on result

File: apps/email-service/src/emails/invite.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+type SendClinicianInviteEmailFailureStatus = SystemError;
+
+/**
+ * Sends a Clinician Invite Email to a Participant
+ * @async
+ * @param {ParticipantClinicianInviteEmailRequest} req - Participant Clinician Invite Email Request
+ * @returns {SentMessageInfo} Sent Message Info
+ */
+export const sendParticipantClinicianInviteEmail = async (
```
Comment (mistryrn) [inline apps/email-service/src/emails/invite.ts:45]: `send<X>Email` functions pull necessary info from props/config, make nodemailer request to send email

File: apps/email-service/src/emails/invite.ts [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+	} catch (error) {
+		logger.error('Unexpected error sending Participant Clinician Invite Email:', error);
+		return failure(
+			'SYSTEM_ERROR',
+			'An unexpected error occurred while sending the Participant Clinician Invite email.',
+		);
+	}
+};
+
+const renderParticipantClinicianInviteEmail = ({
```
Comment (mistryrn) [inline apps/email-service/src/emails/invite.ts:82]: `render<X>Email` functions return HTML compiled from MJML templates via `mjml2html` function. Use string interpolation to insert specific content like names, links, etc

**Review by samrichca — APPROVED** [review]
nice work 🎉

**Pull request merged**